src/main/java/net/glowstone/entity/GlowPlayer.java

```diff
@@ -1643,7 +1644,10 @@ public void setAllowFlight(boolean flight) {
 
     @Override
     public void setFlying(boolean value) {
-        flying = value && canFly;
+        if (flying != (flying = value && canFly)) {
+            EventFactory.getInstance().callEvent(
+                    new PlayerToggleFlightEvent(this, flying));
```

Comment:
This doesn't check if the event is cancelled. The `flying` field should only be updated after the event is fired, and only if the event wasn't cancelled.

Reply:
Thanks for the quick response; that makes sense, I'll fix that up now.